### ISSUE TITLE
Ambiguous Semaphore wait destination scope

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/Vulkan-Docs#1175]
The semaphore wait destination synchronization scope is defined thus:

> The second synchronization scope includes every command submitted in the same batch. In the case of `vkQueueSubmit`, the second synchronization scope is limited to operations on the pipeline stages determined by the destination stage mask specified by the corresponding element of `pWaitDstStageMask`. Also, in the case of `vkQueueSubmit`, the second synchronization scope additionally includes all commands that occur later in submission order.

The wording makes somewhat ambiguous if the limitation of `pWaitDstStageMask` applies to the submission order (i,e, to the last sentence) too. I could interpret it two ways:

a) All commands in the batch limited to `pWaitDstStageMask`. And everything else in submission order. (batch ^ `pWaitDstStageMask`) v submission order

b) The commands in the batch and submission order all limited to `pWaitDstStageMask. (batch v submission order) ^ `pWaitDstStageMask`

**Comment by Tobski** [comment]
Agreed this is ambiguous - and yes the ordering does matter because English language does weird stuff like this 😋 

I think we should just reorder it:

> The second synchronization scope includes every command submitted in the same batch. In the case of vkQueueSubmit, the second synchronization scope additionally includes all commands that occur later in submission order. Also in the case of vkQueueSubmit, the second synchronization scope is limited to operations on the pipeline stages determined by the destination stage mask specified by the corresponding element of pWaitDstStageMask.

A better cleanup would be to not have weird additional queue commands but oh well, here we are 😅

**Comment by krOoze** [comment]
Yea. It is also mildly ambiguous what "later in submission order" refers to. Later than what? The batch? Or the `vkQueueSubmit`?

**Comment by krOoze** [comment]
Now the same unfortunate wording is at semaphore signal op too now that it has `stageMask`.

**Comment by plasmacel** [comment]
I stumbled upon the same issue while interpreting the wording. Is there any update on this?

**Comment by oddhack** [comment]
This should be fixed in the 1.4.338 spec update.
